From eb5b3fdb2338ea55da5e49cb7b1b1da2f4a6f130 Mon Sep 17 00:00:00 2001
From: Vladimir Kostyukov <vladimir.kostyukov@intel.com>
Date: Fri, 7 Jun 2013 15:27:27 +0700
Subject: Dalvik: It should be possible to compile Dalvik with -O0

BZ: 113739

When trying to compile debug version of Dalvik, the compile error
happens due to unresolved sumbol 'ffs' in BitVector.cpp. The thing
is there is a typo in include derective: 'strings.h' instead of
'string.h'. Header 'strings.h' comes from BSD (Mac OS) and it is not
included in default Linux instalation. So, standard 'string.h'header
from 'libc' should be used instead since it's contains all required
routines like 'ffs'. The reason, why this failure was not observed
with default options -O2 is that GCC forwards such calls to builtin
intrinsics. To ___builtin_ffs in our case. Thus, debug version should
use default 'libc' routine while product version should use optimized
intrinsic instead.

Category: device-enablement
Domain: AOSP-Dalvik-Runtime
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: Ie71edd2a4cc8cc75f754b3f3bbd1dd0642a029c2
Orig-MCG-Change-Id: I57db7db9af2b020f2757ad604bacae68e5db3522
Signed-off-by: Vladimir Kostyukov <vladimir.kostyukov@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/BitVector.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vm/BitVector.cpp b/vm/BitVector.cpp
index 635b97d..3d33322 100644
--- a/vm/BitVector.cpp
+++ b/vm/BitVector.cpp
@@ -24,7 +24,7 @@
 #endif
 
 #include <stdlib.h>
-#include <strings.h>
+#include <string.h>
 #include <sstream>
 
 #define kBitVectorGrowth    4   /* increase by 4 u4s when limit hit */
-- 
1.7.4.1

